torch/_export/serde/schema.py

```diff
@@ -0,0 +1,157 @@
+# NOTE: This is a placeholder for iterating on export serialization schema design.
+#       Anything is subject to change and no guarantee is provided at this point.
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+
+class ScalarType(Enum):
+    UNKNOWN = 0
+    BYTE = 1
+    CHAR = 2
+    SHORT = 3
+    INT = 4
+    LONG = 5
+    HALF = 6
+    FLOAT = 7
+    DOUBLE = 8
+    COMPLEXHALF = 9
+    COMPLEXFLOAT = 10
+    COMPLEXDOUBLE = 11
+    BOOL = 12
+    BFLOAT16 = 13
+
+
+class Layout(Enum):
+    Unknown = 0
+    SparseCoo = 1
+    SparseCsr = 2
+    SparseCsc = 3
+    SparseBsr = 4
+    SparseBsc = 5
+    _mkldnn = 6
+    Strided = 7
+
+
+class MemoryFormat(Enum):
+    Unknown = 0
+    ContiguousFormat = 1
+    ChannelsLast = 2
+    ChannelsLast3d = 3
+    PreserveFormat = 4
+
+
+@dataclass
+class Device:
+    type: str
+    index: int
+
+
+@dataclass
+class SymInt:
+    as_symbol: str
+    as_int: int
+
+
+@dataclass
+class TensorMeta:
+    dtype: ScalarType
+    sizes: List[SymInt]
+    requires_grad: bool
+    device: Device
+    strides: Lit[SymInt]
+    storage_offset: int
+    layout: Layout
+
+
+@dataclass
+class SymIntArgument:
+    as_name: str
+    as_int: int
+
+
+@dataclass
+class TensorArgument:
+    name: str
+
+
+# This is actually a union type
+@dataclass
+class Argument:
+    as_none: None
+    as_tensor: TensorArgument
+    as_tensors: List[TensorArgument]
+    as_int: int
+    as_ints: List[int]
+    as_float: float
+    as_floats: List[float]
+    as_string: str
+    as_sym_int: SymIntArgument
+    as_sym_ints: List[SymIntArgument]
+    as_scalar_type: ScalarType
+    as_memory_format: MemoryFormat
+    as_layout: Layout
+    as_device: Device
+    as_bool: bool
+    as_bools: List[bool]
+
+
+@dataclass
+class NamedArgument:
+    name: str
+    arg: Argument
+
+
+@dataclass
+class Operator:
+    name: str
+    version: Optional[int]
+
+
+@dataclass
+class Node:
+    target: Operator
+    inputs: List[NamedArgument]
+    outputs: List[Argument]
+    metadata: Dict[str, str]
+
+
+@dataclass
+class TensorValue:
+    meta: TensorMeta
+
+
+@dataclass
+class Graph:
+    inputs: List[Argument]
+    outputs: List[Argument]
+    nodes: List[Node]
+    tensor_values: Dict[str, TensorValue]
+    sym_int_values: Dict[str, SymIntValue]
+
+
+@dataclass
+class BackwardSignature:
+    gradientsToParameters: Dict[str, str]
+    gradientsToUserInputs: Dict[str, str]
+    lossOutput: str
+
+
+@dataclass
+class GraphSignature:
```

Comment:
Should graph signature also include the pytree in/out?

Reply:
yeah, after some thoughts I think we should contain them.